lib/rspec/bash/call_configuration.rb

```diff
@@ -2,36 +2,65 @@
 
 module Rspec
   module Bash
-    # Configuration of a stubbed command
     class CallConfiguration
       attr_reader :command
 
       def initialize(config_path, command)
         @config_path = config_path
-        @configuration = {}
+        @configuration = []
         @command = command
       end
 
       def set_exitcode(statuscode, args = [])
-        @configuration[args] ||= {}
-        @configuration[args][:statuscode] = statuscode
+        current_conf = create_or_get_conf(args)
+        current_conf[:statuscode] = statuscode
+        write
       end
 
-      def set_output(content, target, args = [])
-        @configuration[args] ||= {}
-        @configuration[args][:outputs] ||= []
-        @configuration[args][:outputs] << { target: target, content: content }
+      def add_output(content, target, args = [])
+        current_conf = create_or_get_conf(args)
+        current_conf[:outputs] << {
+          target: target,
+          content: content
+        }
+        write
       end
 
-      def write
-        structure = @configuration.map do |args, results|
-          { args: args }.merge results
+      def call_configuration
+        return @configuration unless @configuration.empty?
+        begin
+          @config_path.open('r') do |conf_file|
+            YAML.load(conf_file.read) || []
+          end
+        rescue NoMethodError, Errno::ENOENT
+          return []
         end
+      end
+
+      def call_configuration=(new_conf)
+        @configuration = new_conf
+        write
+      end
 
-        @config_path.open('w') do |f|
-          f.puts structure.to_yaml
+      private
+
+      def write
+        @config_path.open('w') do |conf_file|
+          conf_file.write @configuration.to_yaml
         end
       end
+
+      def create_or_get_conf(args)
```

Comment:
I don't know where we landed on whether we thought that the cache was necessary or not